man/fi_peer.3.md

```diff
@@ -192,6 +192,120 @@ on the local node.
    creates a reference between the peer_cq object and its own cq.
 ```
 
+# PEER SRX
+
+The peer SRX defines a mechanism by which peer providers may share a common
+shared receive context.  This avoids the overhead of having separate receive
+queues, can eliminate memory copies, and ensures correct application level
+message ordering.
+
+The setup of a peer SRX is similar to the setup for a peer CQ outlined above.
+A fid_peer_srx object links the owner of the SRX with the peer provider.
+Peer SRXs are configured by the owner calling the peer's fi_srx_context()
+call with the FI_IMPORT_PEER flag set.  The context parameter passed to
+fi_srx_context() must be a struct fi_peer_srx_context.
+
+The data structures to support peer SRXs are defined as follows:
+
+```
+struct fid_peer_srx;
+
+/* Castable to dlist_entry */
+struct fi_peer_rx_entry {
+    struct fi_peer_rx_entry *next;
+    struct fi_peer_rx_entry *prev;
+    fi_addr_t addr;
+    size_t size;
+    uint64_t tag;
+    void *context;
+    size_t iov_cnt;
+    struct iovec iov[];
+};
+
+struct fi_ops_srx_owner {
+    size_t size;
+    int (*get_msg_entry)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+    int (*get_tag_entry)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+};
+
+struct fi_ops_srx_peer {
+    size_t size;
+    int (*start_msg)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+    int (*start_tag)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+    int (*discard_msg)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+    int (*discard_tag)(struct fid_peer_srx *srx,
+        struct fi_peer_rx_entry *entry);
+};
+
+struct fid_peer_srx {
+    struct fid fid;
+    struct fi_ops_srx_owner *owner_ops;
+    struct fi_ops_srx_peer *peer_ops;
+};
+
+struct fi_peer_srx_context {
+    size_t size;
+    struct fid_peer_srx *srx;
+};
+```
+The ownership of structure field values and callback functions is similar
+to those defined for peer CQs, relative to owner versus peer ops.
+
+## fi_ops_srx_owner::get_msg_entry() / get_tag_entry()
```

Comment:
I think `get` is too general. Maybe a different word like `claim`?

Reply:
Claim has a specific meaning relative to the main API.  We're either getting a receive buffer (expected case) or queuing a request for a receive buffer (unexpected case).